livingdoc-engine/src/main/kotlin/org/livingdoc/engine/LivingDoc.kt

```diff
@@ -60,14 +57,52 @@ class LivingDoc(
         }
     }
 
+    private fun executableDecisionTable(element: DecisionTable, documentModel: ExecutableDocumentModel): ExecutableDecisionTable? {
+        val matchingFixture = decisionTableToFixtureMatcher.findMatchingFixture(element, documentModel.decisionTableFixtures)
+        if (matchingFixture != null) {
+            return ExecutableDecisionTable(element) {
+                decisionTableExecutor.execute(it, matchingFixture, documentModel.documentClass)
+            }
+        }
+        return null
+    }
+
+    private fun executableScenario(scenario: Scenario, documentModel: ExecutableDocumentModel): ExecutableScenario? {
+        val matchingFixture = scenarioToFixtureMatcher.findMatchingFixture(scenario, documentModel.decisionTableFixtures)
+        if (matchingFixture != null) {
+            return ExecutableScenario(scenario) {
+                scenarioExecutor.execute(it, matchingFixture, documentModel.documentClass)
+            }
+        }
+        return null
+    }
+
+}
+
+sealed class ExecutableExample<A : Example, B : ExampleResult>(
```

Comment:
You've been pushing for keeping this class small and straightforward, now it's filled with private sealed companions. We should discuss how to take this apart again.